aspnet/fundamentals/static-files.rst

```diff
@@ -1,8 +1,127 @@
-.. include:: /../common/stub-topic.txt
+Working with Static Files
+=========================
+By `Tom Archer`
 
-|stub-icon| Working with Static Files
-=====================================
+Static files, which include HTML files, CSS files, image files, and JavaScript files, are assets that the app will serve directly to clients. In this article, we'll cover the following topics as they relate to ASP.NET 5 and static files.
 
-.. include:: /../common/stub-notice.txt
+In this article:
+  - `Serving static files`_
+  - `Enabling directory browsing`_
+  - `Serving default files`_
+  - `Using the UseFileServer method`_
+  - `Best practices`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/60
\ No newline at end of file
+Serving static files
+--------------------
+
+By default, static files are stored in the `webroot` of your project. The location of the webroot is defined in the project's ``project.json`` file where the default is `wwwroot`.
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/project.json
+  :linenos:
+  :language: json
+  :lines: 2
+
+Static files can be stored in any folder under the webroot and accessed with a relative path to that root. For example, when you create a default Web application project using Visual Studio, there are several folders created within the webroot folder - ``css``, ``images``, and ``js``. In order to directly access an image in the ``images`` subfolder, the URL would look like the following:
+
+  http://<yourApp>/images/<imageFileName>
+
+In order for static files to be served, you must configure the :doc:`middleware` to add static files to the pipeline. This is accomplished by calling the  ``UseStaticFiles`` extension method from  ``Startup.Configure`` as follows:
+
+.. code-block:: c#
+  :emphasize-lines: 5
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Add static files to the request pipeline.
+    app.UseStaticFiles();
+    ...
+
+Enabling directory browsing
+---------------------------
+
+Directory browsing allows the user of your Web app to see a list of directories and files within a specified directory (including the root). By default, this functionality is turned off such that if the user attempts to display a directory within an ASP.NET Web app, the browser displays an error. To enable directory browsing for your Web app, call the ``UseDirectoryBrowser`` extension method from  ``Startup.Configure`` as follows:
```

Comment:
"By default, this functionality is turned off" -> "is not available"?  Middleware does not enable existing functionality, it adds new functionality to the application.